<launch>

	<arg name="namespace" default="Sauron"/>
	<arg name="rviz_cfg"  default="-d $(find px4_control)/Extras/$(arg namespace)Inspection.rviz" />
    <arg name="launch_rviz" default="true" />

    <group ns="$(arg namespace)">

        <!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
	    <node pkg="mission_planner" type="vicon_rover_inspection" name="inspector_node" output="screen">

	    	<!-- Namespace -->
	    	<param name="namespace" value="$(arg namespace)"/>

			<!-- Paths for waypoint files -->
			<param name="inspection_waypoints_path" value="$(find mission_planner)/waypoints/wps_r200.txt"/>

			<!-- Guidance parameters -->
			<param name="takeoff_height" value="0.6"/> <!-- Takeoff height (in meters) -->
			<param name="max_velocity" value="0.15"/>   <!-- Maximum speed (in meters/sec) -->

			<!-- Thread update rates (in Hz) -->
			<param name="tf_update_rate" value="100"/>

			<!-- Whether or not to save octomap -->
			<param name="save_octomap" value="false"/>
		</node>

		<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
    	<node pkg="mav_trajectory_generation_ros" type="minSnap_Node" name="minSnap_Node" output="screen"/> 

    </group>

	<!-- Batch solver to find relative pose between vicon and SLAM -->
    <include file="$(find batch_pose_estimator)/launch/slam2vicon.launch">
	    <arg name="namespace" value="$(arg namespace)" />
    </include>

    <!-- Neural network -->
    <include file="$(find darknet_ros)/launch/symbols.launch"> 
        <arg name="namespace" value="$(arg namespace)" />
    </include>

    <!-- Triangulation node to find emoji locations -->
    <include file="$(find yolo_triangulation)/launch/triangulate.launch">
	    <arg name="namespace" value="$(arg namespace)" />
    </include>

    <!-- Mapper node -->
    <!-- <include file="$(find mapper)/launch/mapper.launch">
		<arg name="namespace" value="$(arg namespace)" />
		<arg name="launch_rviz" value="false" />
		<arg name="process_pcl_at_startup" value="false" />
		<arg name="inertial_frame_id" default="/slam" />
		<arg name="depth_cam_suffix" default="/camera/depth/compressed_points" />
	</include> -->

    <!-- <include file="$(find mapper)/launch/mapper_ns.launch">
		<arg name="namespace" value="$(arg namespace)" />
	</include> -->

    <!-- Launch RVIZ -->
    <!-- <group if="$(arg launch_rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="$(arg rviz_cfg)"/>
    </group> -->


</launch>